tests/Workflow/Integration/Test_UserJobs.py

```diff
@@ -27,9 +28,16 @@ def setUp(self):
     super(UserJobTestCase, self).setUp()
 
     self.d = Dirac()
-    self.exeScriptLocation = find_all('exe-script.py', rootPath, '/DIRAC/tests/Workflow')[0]
-    self.helloWorld = find_all("helloWorld.py", rootPath, '/DIRAC/tests/Workflow')[0]
-    self.mpExe = find_all('testMpJob.sh', rootPath, '/DIRAC/tests/Utilities')[0]
+
+    try:
+      self.exeScriptLocation = find_all('exe-script.py', rootPath, '/DIRAC/tests/Workflow')[0]
+      self.helloWorld = find_all("helloWorld.py", rootPath, '/DIRAC/tests/Workflow')[0]
+      self.mpExe = find_all('testMpJob.sh', rootPath, '/DIRAC/tests/Utilities')[0]
+    except IndexError:  # we are in Jenkins
```

Comment:
Why is it different for Jenkins ? 
In any case, the 3 lines look strictly identical to me except the middle variable name. Can't you do a test before and assign a variable to either `rootPath` or `os.environ['WORKSPACE']` so you do not repeat the code ?

Reply:
No, `rootPath`is the directory where DIRAC is installed, while I need `os.environ['WORKSPACE']` because in Jenkins the tests are not necessarily found in the directory where DIRAC is installed. 

For example: the pilot installs DIRAC in a certain directory, so it fixes its `rootPath`, but the tests are in fact found in a separate directory. I would not mix the 2.